### PR TITLE
Fetch recent patients Part 1

### PR DESCRIPTION
**Pull request by rakshakhegde**
https://www.pivotaltracker.com/story/show/163976144

Database fetching is completely working!

### CODE REVIEW

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 1]
```diff
@@ -874,4 +875,104 @@ class PatientRepositoryAndroidTest {
...
+  }
+
+  private fun Patient.toRecentPatient(bpMeasurement: BloodPressureMeasurement) = RecentPatient(
+      uuid = uuid,
+      fullName = fullName,
+      gender = gender,
+      dateOfBirth = dateOfBirth,
+      age = age,
+      bpUpdatedAt = bpMeasurement.updatedAt,
+      lastBp = RecentPatient.LastBp(bpMeasurement.systolic.toString(), bpMeasurement.diastolic.toString())
```
Comment (sanchita) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt]: `LastBp` class should also carry the date when BP was taken.
You will have to display that in `RecentPatientItem`

Comment (sanchita) [reply app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt]: Also, I think `LastBp` should take `systolic` and `diastolic` as `Int`.
The conversion to `String` should happen in the views.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 2]
```diff
@@ -21,6 +21,7 @@ import org.simple.clinic.facility.FacilityRepository
 import org.simple.clinic.medicalhistory.MedicalHistoryRepository
```
Comment (sanchita) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt]: I can see that this commit is doing more than adding a test.
Can you separate out the changes to the query in a different commit?

File: app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt [hunk 3]
```diff
@@ -32,6 +32,11 @@ data class RecentPatient(
   @Dao
   interface RoomDao {
 
+    /**
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt]: Can you mention what "RecentPatient" means first and then describe the query?

Comment (sanchita) [reply app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt]: Also, it is alright if the query is not described at all. That is more of a good to have.

File: app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt [hunk 4]
```diff
@@ -38,7 +38,10 @@ data class RecentPatient(
         MAX(
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt]: Typo?
`BP.updatedAt bpUpdatedAt` -> `bp_updatedAt`

File: app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt [hunk 5]
```diff
@@ -53,9 +56,26 @@ data class RecentPatient(
...
+              GROUP BY appointmentUuid
+          ) COMM ON AP.uuid = COMM.appointmentUuid
+          LEFT JOIN (
+            SELECT MAX(updatedAt) latestUpdatedAt, T.*
+              FROM MedicalHistory T
+              GROUP BY patientUuid
+          ) MH ON P.uuid = MH.patientUuid
         WHERE (
           BP.facilityUuid = :facilityUuid OR
-          PD.facilityUuid = :facilityUuid
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt]: Do we even need this `WHERE` clause ?
We are already checking `facilituUuid` in individual `LEFT JOIN` queries.
Please add a test to verify this.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt (hunk 1 above)
Comment (rakshakhegde) [reply app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt]: Will be displaying the date as part of next PR.

File: app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt (hunk 5 above)
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt]: Added test, seems like this WHERE clause is required. Need to investigate why.

File: app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt (hunk 3 above)
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/patient/recent/RecentPatient.kt]: I've updated the comment.

**Pull request merged**